astropy/table/tests/test_mixin.py

```diff
@@ -114,11 +114,16 @@ def test_io_ascii_write():
     every pure Python writer.  No validation of the output is done,
     this just confirms no exceptions.
     """
+    import warnings
+
     from astropy.io.ascii.connect import _get_connectors_table
+    from astropy.io.ascii.tdat import TdatFormatWarning
```

Comment:
I think it would be better to special-case this format and set `.meta["table_name"]` in the table. Suppressing these warnings could be masking some other warning since there are a number of ways `TdatFormatWarning` can be raised.